src/client/Microsoft.Identity.Client/Account.cs

```diff
@@ -20,25 +20,30 @@ internal sealed class Account : IAccount
         /// <param name="username">UPN style , can be null</param>
         /// <param name="environment">Identity provider for this account, e.g. <c>login.microsoftonline.com</c></param>
         /// <param name="wamAccountIds">Map of (client_id, wam_account_id)</param>
-        public Account(string homeAccountId, string username, string environment,IDictionary<string, string> wamAccountIds = null)
+        /// <param name="tenantProfiles">Map of (tenant_id, tenant_profile)</param>
+        public Account(
+            string homeAccountId, 
+            string username, 
+            string environment, 
+            IDictionary<string, string> wamAccountIds = null, 
+            IDictionary<string, TenantProfile> tenantProfiles = null)
         {
             Username = username;
             Environment = environment;
             HomeAccountId = AccountId.ParseFromString(homeAccountId);
             WamAccountIds = wamAccountIds;
+            TenantProfiles = tenantProfiles;
         }        
 
         public string Username { get; }
 
         public string Environment { get; }
 
         public AccountId HomeAccountId { get; }
-       
 
-        internal IDictionary<string, string> WamAccountIds
-        {
-            get;
-        }
+        public IDictionary<string, TenantProfile> TenantProfiles { get; }
```

Comment:
Should this also be `IEnumerable<TenantProfile>` for consistency?

Reply:
Internally, we construct it as a dictionary, to avoid duplicates. So it is consistent with MSAL internals.